File: Java/src/Binary Tree Level Order Traversal.java
/**
 * Definition for binary tree
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public List<List<Integer>> levelOrder(TreeNode root) {
        List<List<Integer>> ret = new ArrayList<List<Integer>>();
        helper(root, 1 , ret);
        return ret;
    }
    
    public void helper(TreeNode root, int level, List<List<Integer>> ret){
        if (root == null) {
            return;
        }
        
        if (ret.size() < level){
            ArrayList<Integer> item = new ArrayList<Integer>();
            item.add(root.val);
            ret.add(item);
        }
        else{
            ret.get(level-1).add(root.val);
        }
        
        helper(root.left, level+1, ret);
        helper(root.right, level+1, ret);
    }
}

File: Python/Binary Tree Level Order Traversal.py
class Solution:
    def levelOrder(self, root):
        ret = []
        self.helper(root,1,ret)
        return ret

    def helper(self, root, level, ret):
        if not root:
            return
        if len(ret) < level:
            ret.append([root.val])
        else:
            ret[level-1].append(root.val)
        self.helper(root.left, level+1, ret)
        self.helper(root.right, level+1, ret)